### PR TITLE
add support for metadata object removal

### PR DESCRIPTION
**Pull request by sadam21**

### CODE REVIEW

File: src/main/java/com/gooddata/md/MetadataService.java [hunk 1]
```diff
@@ -94,6 +94,56 @@ public MetadataService(RestTemplate restTemplate) {
     }
 
     /**
+     * Remove metadata object URI
+     *
+     * @param obj metadata object to remove
+     * @param <T> type of the object to be returned
+     * @return the metadata object
```
Comment (pirkogdc) [inline src/main/java/com/gooddata/md/MetadataService.java]: it returns void, so this is redundant

File: src/main/java/com/gooddata/md/MetadataService.java [hunk 2]
```diff
@@ -94,6 +94,56 @@ public MetadataService(RestTemplate restTemplate) {
...
+        try {
+            restTemplate.delete(obj.getUri());
+        } catch (GoodDataRestException e) {
+            if (HttpStatus.NOT_FOUND.value() == e.getStatusCode()) {
+                throw new ObjNotFoundException(obj);
+            } else {
+                throw e;
+            }
+        } catch (RestClientException e) {
+            throw new GoodDataException("Unable to remove " + obj.getClass().getSimpleName().toLowerCase() + " " + obj.getUri(), e);
```
Comment (pirkogdc) [inline src/main/java/com/gooddata/md/MetadataService.java]: we should consider add toString to `AbstractObj`

File: src/main/java/com/gooddata/md/MetadataService.java [hunk 3]
```diff
@@ -94,6 +94,56 @@ public MetadataService(RestTemplate restTemplate) {
...
+        }
+    }
+
+    /**
+     * Remove metadata object by URI (format is <code>/gdc/md/{PROJECT_ID}/obj/{OBJECT_ID}</code>)
+     *
+     * @param uri URI in format <code>/gdc/md/{PROJECT_ID}/obj/{OBJECT_ID}</code>
+     * @param <T> type of the object to be returned
+     * @return the metadata object
```
Comment (pirkogdc) [inline src/main/java/com/gooddata/md/MetadataService.java]: it returns void, so this is redundant

File: src/main/java/com/gooddata/md/MetadataService.java (hunk 1 above)
Comment (sadam21) [inline src/main/java/com/gooddata/md/MetadataService.java]: Javadoc fixed.

File: src/main/java/com/gooddata/md/MetadataService.java (hunk 3 above)
Comment (sadam21) [inline src/main/java/com/gooddata/md/MetadataService.java]: Javadoc fixed.

File: src/main/java/com/gooddata/md/MetadataService.java (hunk 2 above)
Comment (sadam21) [inline src/main/java/com/gooddata/md/MetadataService.java]: OK, but I would do it in another pull request.

File: src/main/java/com/gooddata/md/MetadataService.java [hunk 4]
```diff
@@ -94,6 +94,54 @@ public MetadataService(RestTemplate restTemplate) {
...
     /**
+     * Remove metadata object URI
+     *
+     * @param obj metadata object to remove
+     * @param <T> type of the object to be returned
+     * @throws com.gooddata.md.ObjNotFoundException if metadata object not found
+     * @throws com.gooddata.GoodDataRestException   if GoodData REST API returns unexpected status code
+     * @throws com.gooddata.GoodDataException       if no response from API or client-side HTTP error
+     */
+    public <T extends Obj> void removeObj(T obj) {
```
Comment (martiner) [inline src/main/java/com/gooddata/md/MetadataService.java]: The type info is redundant `public void removeObj(Obj obj)` will do the same trick

Comment (sadam21) [inline src/main/java/com/gooddata/md/MetadataService.java]: You are right, I'll remove it.

File: src/main/java/com/gooddata/md/ObjNotFoundException.java [hunk 5]
```diff
@@ -22,6 +22,25 @@
     /**
      * Construct a new instance of ObjNotFoundException.
      *
+     * @param obj metadata object you're working with
+     * @param <T> the type of results you're searching for
+     */
+    public <T extends Obj> ObjNotFoundException(T obj) {
```
Comment (martiner) [inline src/main/java/com/gooddata/md/ObjNotFoundException.java]: redundant type parameter

Comment (sadam21) [inline src/main/java/com/gooddata/md/ObjNotFoundException.java]: Removed...
